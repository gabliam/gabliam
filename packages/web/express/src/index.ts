import { ExpressPlugin } from './express-plugin';

export * from 'http-status-codes';
export * from './express';
export * from './express-converter';
export * from './express-interceptor';
export * from './interfaces';
export * from './request';
export * from './response';
export { ExpressPlugin as default };
