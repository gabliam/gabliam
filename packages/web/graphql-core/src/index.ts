export { TYPE } from './constants';
export * from './graphql-core-plugin';
export * from './interfaces';
export * from './resolver-decorator';
export {
  Arg,
  ArgOptions,
  Args,
  ArgsType,
  Authorized,
  createParamDecorator,
  createMethodDecorator,
  Ctx,
  Directive,
  Extensions,
  registerEnumType,
  Field,
  FieldOptions,
  FieldResolver,
  Info,
  InputType,
  InputTypeOptions,
  InterfaceType,
  InterfaceTypeOptions,
  Mutation,
  ObjectType,
  ObjectTypeOptions,
  PubSub,
  Query,
  Resolver,
  Root,
  Subscription,
  SubscriptionOptions,
  createUnionType,
  UseMiddleware,
  ID, Int, ClassType, Publisher
} from 'type-graphql';
