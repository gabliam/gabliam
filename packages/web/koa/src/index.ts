import { KoaPlugin } from './koa-plugin';

export * from 'http-status-codes';
export * from './interfaces';
export * from './koa';
export * from './koa-converter';
export * from './koa-interceptor';
export * from './request';
export * from './response';
export { KoaPlugin as default };
