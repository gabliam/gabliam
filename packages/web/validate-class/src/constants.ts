export const METADATA_KEY = {
  validate: 'validate-class_validate',
};

export const ERRORS_MSGS = {
  DUPLICATED_VALIDATE_DECORATOR: `Cannot apply @Validate decorator multiple times.`,
};
