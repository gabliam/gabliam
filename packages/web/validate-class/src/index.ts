export * from './constants';
export * from './metadatas';
export * from './validate';
