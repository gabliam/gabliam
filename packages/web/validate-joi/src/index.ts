export * from './metadatas';
export * from './validate';
export * from './constants';
export * from './validate-config';
export { ValidatePlugin as default } from './validate-plugin';
