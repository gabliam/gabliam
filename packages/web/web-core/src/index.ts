export * from './constants';
export * from './cookies';
export * from './errors';
export * from './exceptions';
export * from './execution-context';
export * from './gab-context';
export * from './gab-request';
export * from './gab-response';
export * from './interceptor';
export * from './interface';
export * from './metadatas';
export * from './pipe';
export * from './plugin-config';
export * from './response-entity';
export * from './server-starter';
export * from './utils';
export * from './web-configuration';
export * from './web-plugin-base';
