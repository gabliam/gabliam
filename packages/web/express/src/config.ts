import { Bean, Joi, PluginConfig, Value } from '@gabliam/core';
import bodyParser from 'body-parser';
import { EXPRESS_PLUGIN_CONFIG } from './constants';

const options = Joi.object().keys({
  inflate: Joi.boolean(),
  limit: [Joi.number(), Joi.string()],
  type: [Joi.string(), Joi.array().items(Joi.string())],
});

const optionsJson = options.keys({
  strict: Joi.boolean(),
});

const optionsText = options.keys({
  defaultCharset: Joi.string(),
});

const optionsUrlencoded = options
  .keys({
    extended: Joi.boolean(),
    parameterLimit: Joi.number(),
  })
  .default({ extended: false });

const configJoi = Joi.object()
  .keys({
    json: Joi.boolean().default(true),
    optionsJson,
    urlencoded: Joi.boolean().default(true),
    optionsUrlencoded,
    text: Joi.boolean().default(false),
    optionsText,
  })
  .default({
    json: true,
    urlencoded: true,
    optionsUrlencoded: { extended: false },
    text: false,
  });

export interface BodyParserConfig {
  json: boolean;
  optionsJson?: bodyParser.OptionsJson;
  urlencoded: boolean;
  optionsUrlencoded?: bodyParser.OptionsUrlencoded;
  text: boolean;
  optionsText?: bodyParser.OptionsText;
}

export interface ExpressConfig {
  bodyParser: BodyParserConfig;
}

@PluginConfig()
export class ExpressPluginConfig implements ExpressConfig {
  @Value('application.web.bodyParser', configJoi)
  bodyParser: BodyParserConfig;

  @Bean(EXPRESS_PLUGIN_CONFIG)
  restConfig(): ExpressConfig {
    return {
      bodyParser: this.bodyParser,
    };
  }
}
