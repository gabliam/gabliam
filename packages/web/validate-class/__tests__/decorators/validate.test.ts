import { reflection } from '@gabliam/core';
import { Controller, Get } from '@gabliam/web-core';
import { Validate } from '../../src';

describe('Validate decorator', () => {
  test('should add Validate metadata to a method when decorated with @Validate', () => {
    @Controller('/test')
    class TestController {
      @Get(`/`)
      @Validate()
      test() {
        return '';
      }
    }

    expect(reflection.propMetadata(TestController)).toMatchSnapshot();
  });
  test('should add Validate metadata to a method when decorated with @Validate(validatorOptions)', () => {
    @Controller('/test')
    class TestController {
      @Get(`/`)
      @Validate({
        transform: true,
      })
      test() {
        return;
      }
    }

    expect(reflection.propMetadata(TestController)).toMatchSnapshot();
  });

  test('should fail when decorated multiple times the same method with @Validate', () => {
    expect(function() {
      @Controller('/test')
      class TestController {
        @Get(`/`)
        @Validate()
        @Validate()
        test() {
          return;
        }
      }
      // tslint:disable-next-line:no-unused-expression
      new TestController();
    }).toThrowError();
  });
});
