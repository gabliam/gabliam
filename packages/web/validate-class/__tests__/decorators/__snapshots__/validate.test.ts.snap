// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate decorator should add Validate metadata to a method when decorated with @Validate 1`] = `
{
  "test": [
    PropDecoratorFactory {
      "method": "get",
      "path": "/",
    },
    DecoratorFactory {
      "ids": [
        ValidatePipe {
          "options": DecoratorFactory {
            "disableErrorMessages": false,
            "exceptionFactory": [Function],
            "transform": false,
            "transformOptions": {},
            "validatorOptions": {
              "validationError": {
                "target": false,
              },
            },
          },
        },
      ],
    },
    DecoratorFactory {
      "disableErrorMessages": false,
      "exceptionFactory": [Function],
      "transform": false,
      "transformOptions": {},
      "validatorOptions": {
        "validationError": {
          "target": false,
        },
      },
    },
  ],
}
`;

exports[`Validate decorator should add Validate metadata to a method when decorated with @Validate(validatorOptions) 1`] = `
{
  "test": [
    PropDecoratorFactory {
      "method": "get",
      "path": "/",
    },
    DecoratorFactory {
      "ids": [
        ValidatePipe {
          "options": DecoratorFactory {
            "disableErrorMessages": false,
            "exceptionFactory": [Function],
            "transform": true,
            "transformOptions": {},
            "validatorOptions": {
              "validationError": {
                "target": false,
              },
            },
          },
        },
      ],
    },
    DecoratorFactory {
      "disableErrorMessages": false,
      "exceptionFactory": [Function],
      "transform": true,
      "transformOptions": {},
      "validatorOptions": {
        "validationError": {
          "target": false,
        },
      },
    },
  ],
}
`;
